core/node/house_keeper/src/fri_witness_generator_jobs_retry_manager.rs

```diff
@@ -88,6 +88,28 @@ impl FriWitnessGeneratorJobRetryManager {
         );
     }
 
+    pub async fn requeue_stuck_recursion_tip_jobs(&mut self) {
+        let stuck_jobs = self
+            .pool
+            .connection()
+            .await
+            .unwrap()
+            .fri_witness_generator_dal()
+            .requeue_stuck_recursion_tip_jobs(
+                self.processing_timeouts.recursion_tip(),
+                self.max_attempts,
+            )
+            .await;
+        let job_len = stuck_jobs.len();
+        for stuck_job in stuck_jobs {
+            tracing::info!("re-queuing fri witness input job {:?}", stuck_job);
+        }
+        metrics::counter!(
```

Comment:
I would ask you to use vise for metrics, will be much easier to track them. Just put the metrics in metrics.rs file. I know that we don't have it like that in house keeper, but I think we'll change that.